### PR TITLE
palemoon: 27.9.4 -> 28.6.0.1, refactor

### LINKED ISSUE #51160: [Security Update] Palemoon 28.2.1
## Issue description
On November 13th Palemoon 28.2.0 was released with very important security updates: http://www.palemoon.org/releasenotes.shtml. This browser has an integrated update checker which is complaining repeatedly about the security update (which is good!) 

### Steps to reproduce

Install palemoon, wait a few minutes, a pop-up window raises asking to update palemoon.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.4.0-139-generic, Ubuntu, 14.04.5 LTS, Trusty Tahr`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1.3`
 - channels(user): `"nixpkgs-19.03pre160446.206a1c00bae"`
 - nixpkgs: `/home/user/.nix-defexpr/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by OPNA2608**
supersedes #51161
closes #51160

Screenshot_NixOS_19 03_2019-07-09_08:36:04

---

GConf dependency added as discussed in #55161.

The NixOS build problem that was encountered there took some log digging to find:
There was an attempted invocation of `m4` that failed, but didn't causes an abortion of the build phase. Later on, that seemingly resulted in some files missing and an abortion of the install phase.

This seems to have not been an issue on non-NixOS systems, where the lack of sandboxing lead to `m4` being found inside the user's PATH?

The configure script also checks for the presence of `wget`, but I don't know what for. Just to be safe, I added it to `buildInputs` as well.

I think `gstreamer` was deprecated afew versions ago and can be removed from `buildInputs`, I will do some testing in general to figure out what else might not be required anymore though.

(Also, it looks like Pale Moon can be built for GTK3 so having a `withGtk3` option could make sense, question would be how endorsed that is by upstream and if it may continue using the official branding in that case.)

---

~~A major *not-nice* aspect of this PR is a set of exceptions that get thrown during the install phase:~~
~~They don't abort the installation however, and the browser appears to work fine. I don't know how to fix those yet, but I'll look into it. If anyone wants to help me, be my guest ofc. :slightly_smiling_face:~~

See comment below.

###### Motivation for this change

see #51160

tl;dr, the currently packaged version is

1) out-of-date
2) frequently complaining about being out-of-date
3) lacking important security updates that have been added since its release

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by OPNA2608** [comment]
I think those exceptions are *"fine"?* Their [fully-endorsed Fedora builds](https://copr-be.cloud.fedoraproject.org/results/bgstack15/palemoon/fedora-28-x86_64/00965175-palemoon/) throw the exact same errors in their build log, with branding and all.

I still want to work on the build inputs abit and see what's redundant, but it should otherwise be done imo.

**Comment by OPNA2608** [comment]
I've removed gstreamer, as support for it it was deprecated and removed some updates ago.

Some other dependencies like nss should be unused because Pale Moon bundles specific versions of these. They should be replaceable with the system's versions of them, but upstream does not like that - especially not with official branding. Hence, I removed as many as I could identify so far. I haven't noticed any difference so far, so all seems to be well.

I've added ffmpeg and wrapped the final executable's LD_LIBRARY_PATH so it can access it. YouTube's HTML5 test page is happy and playback of 60Hz content works, but it's noticeably less smooth than Firefox'. I'll check if I can fix that.

---

I'll use this at home and comment on / fix any problems I might have, but so far it seems to work well. Compilation took abit more than 30min on a VM with 3 cores to run the build on.

I'd appreciate it if someone with the right permissions could kick off a build on @GrahamcOfBorg and adjust the timeout if neccessary :slightly_smiling_face:

**Comment by OPNA2608** [comment]
I can't figure out what might be causing the unsmooth 60Hz YouTube video playback, perhaps it's just my computer being too weak.

That being said, this PR should be done. I've been using it for the past few days and it works fine.

**Comment by OPNA2608** [comment]
ping @kalbasit, because he reviewed the old PR that's superseded by this

**Comment by kalbasit** [comment]
@GrahamcOfBorg build palemoon

**Review by kalbasit — APPROVED** [review]
I built Palemoon locally on my NixOS box and it works very well. I have not tested on a different Linux box though. Also it looks like the tests are failing on aarch64-linux.

**Comment by OPNA2608** [comment]
> Also it looks like the tests are failing on aarch64-linux.

I don't think upstream supports non-x86 architectures or macOS, so I can change the platforms.
```nix
platforms = [ "i686-linux" "x86_64-linux" ];
```
